### PR TITLE
Make multiplexing work properly by hooking up viewlets

### PR DESCRIPTION
**Pull request by shans**
instead of views as return value of createView.

### CODE REVIEW

File: runtime/test/particle-api-test.js [hunk 1]
```diff
@@ -280,7 +284,8 @@ describe('particle-api', function() {
     recipe.normalize();
     arc.instantiate(recipe);
 
-    await util.assertViewWillChangeTo(resultsView, Result, "value", ["done", "done"]);
+    await util.assertViewWillChangeTo(resultsView, Result, "value", ["done", "done", "HELLO", "WORLD"]);
```
Comment (mmandlis) [inline runtime/test/particle-api-test.js]: i think it is better to get rid of "done" here (have the particle not put it in the results view).

File: runtime/test/test-util.js [hunk 2]
```diff
@@ -42,9 +42,11 @@ function assertViewWillChangeTo(setView, entityClass, field, expectations) {
     view.on('change', () => view.toList().then(result => {
       if (result == undefined)
         return;
-      if (result.length == expectations.length &&
-          result.every(a => expectations.indexOf(a[field]) >= 0)) {
-        resolve();
+      if (result.length == expectations.length) {
```
Comment (mmandlis) [inline runtime/test/test-util.js]: thanks for fixing this! i figured this cannot be right, but didn't know how to make it better :)

File: runtime/test/particle-api-test.js (hunk 1 above)
Comment (shans) [reply runtime/test/particle-api-test.js]: actually I think it's kinda cool that this works :)

I'm quite OK to remove it to make the test cleaner, but it's also a nice demonstration that both the transformation particle and contained particles can work on the same output views.

Comment (mmandlis) [reply runtime/test/particle-api-test.js]: i guess, you're right!
more elegant particle purpose is not as important here as demonstration of capabilities. so let's leave it as is.

thanks!

Comment (mmandlis) [reply runtime/test/particle-api-test.js]: actually, a question. when we do in a loop:

let res = resultsView.store(new resultsView.entityClass({value: 'done'}));
outView.on('change', () => {
    outView.get().then(result => {
        if (result) {
            resultsView.store(result);
        }
   })
}, this);

is it possible that the first outView change would happen before the next "done" is added? could the resultsView end up being
['done', 'HELLO', 'done', 'WORLD'] ?

**Pull request merged**